clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1280,42 +1276,36 @@ protected void produceToStoreBufferServiceOrKafka(
     }
   }
 
-  private void drainExceptionQueue(BlockingQueue<PartitionExceptionInfo> exceptionQueue, String exceptionGroup) {
-    while (!exceptionQueue.isEmpty()) {
-      try {
-        PartitionExceptionInfo exceptionPartitionPair = exceptionQueue.take();
-        int exceptionPartition = exceptionPartitionPair.getPartitionId();
-        PartitionConsumptionState partitionConsumptionState = partitionConsumptionStateMap.get(exceptionPartition);
-        if (partitionConsumptionState == null) {
-          LOGGER.warn(
-              "Ignoring exception for partition {} for store version {} since this partition has been unsubscribed already.",
-              exceptionPartition,
-              kafkaVersionTopic,
-              exceptionPartitionPair.getException());
-          continue;
-        }
+  private void processIngestionException(List<Exception> exceptionList, String exceptionGroup) {
+    for (int exceptionPartition = 0; exceptionPartition < exceptionList.size(); ++exceptionPartition) {
```

Comment:
minor perf improvement: may be iterate over errorPartitions keys instead of the list.

Reply:
Removed `errorPartitions` to avoid potential race condition.

Reply:
I see, any reason to use SparseConcurrentList over a Map<Int, Exception> ?

Reply:
It is faster since it is backed by a list.

Reply:
Since its not on hot path, it wont matter.  consider using SparseConcurrentList::forEach to remove boiler plate null check.